heka/sandbox/filters/firefox_duplicates.lua

```diff
@@ -0,0 +1,74 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+Firefox Duplicate Telemetry Submission Report
+
+*Example Heka Configuration*
+
+.. code-block:: ini
+
+    [FirefoxDuplicates]
+    type = "SandboxFilter"
+    filename = "lua_filters/firefox_duplicates.lua"
+    message_matcher = "Logger == 'fx' && Type == 'executive_summary'"
+    output_limit = 0
+    memory_limit = 0
+    instruction_limit = 0
+    ticker_interval = 0
+    preserve_data = false
+    timer_event_on_shutdown = true
+
+        [FirefoxDuplicates.config]
+        items = 100000000
+--]]
+
+require "bloom_filter"
+require "circular_buffer"
+local fx = require "fx"
+
+local items = read_config("items") or 1000000
+local probability = read_config("probability") or 0.01
+bf = bloom_filter.new(items, probability)
```

Comment:
What will happen if you try to create a bloom filter that requires more memory than is available?